### PR TITLE
Hibernate task solution

### PR DESCRIPTION
**Pull request by mkazarian-1**

### CODE REVIEW

File: src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        <property name="hbm2ddl.auto">create</property>
+
+        <mapping class="mate.academy.model.Movie"/>
+        <mapping class="mate.academy.model.MovieSession"/>
+        <mapping class="mate.academy.model.CinemaHall"/>
+        <mapping class="mate.academy.model.ShoppingCart"/>
+        <mapping class="mate.academy.model.Ticket"/>
+        <mapping class="mate.academy.model.User"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (okuzan) [inline src/main/resources/hibernate.cfg.xml:18]:
```suggestion
</hibernate-configuration>

```

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        shoppingCart.getTickets().add(ticket);
+
+        ticketDao.add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user)
+                .orElseThrow(() -> new RuntimeException("Can't get shoppingCart by user" + user));
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java:36]: EntityNotFoundException is more suitable here
use id in the message, not whole user

File: src/main/java/mate/academy/model/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
```
Comment (okuzan) [inline src/main/java/mate/academy/model/Ticket.java:17]: add LAZY + JoinColumn

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/resources/hibernate.cfg.xml (hunk 1 above)
Comment (mkazarian-1) [reply src/main/resources/hibernate.cfg.xml:18]: Please explain the problem in more details

What should i change in configuration file

Comment (Elena-Bruyako) [reply src/main/resources/hibernate.cfg.xml:18]: mentor suggested to add empty line at the end of file

File: src/main/java/mate/academy/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    @OneToMany
+    private List<Ticket> tickets;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @MapsId
+    private User user;
+
+    @Override
+    public String toString() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/ShoppingCart.java:26]: replace after get and set methods

File: src/main/java/mate/academy/model/Ticket.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    private MovieSession movieSession;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private User user;
+
+    @Override
+    public String toString() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Ticket.java:24]: same

File: src/main/java/mate/academy/service/ShoppingCartService.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package mate.academy.service;
+
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+
+public interface ShoppingCartService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/ShoppingCartService.java:8]:
```suggestion
```

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+
+        ShoppingCart shoppingCart = this.getByUser(user);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java:27]:
```suggestion
        ShoppingCart shoppingCart = getByUser(user);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request closed without merging**